src/variables.ts

```diff
@@ -0,0 +1,314 @@
+import type { CompanionVariableDefinition } from '@companion-module/base'
+import { playerTimeToString, type SoundcraftUI } from 'soundcraft-ui-connection'
+import { firstValueFrom, map, Observable } from 'rxjs'
+
+import type { UiVariablesStore } from './variables-store.js'
+
+/** Map -Infinity to -100 for fader level representation */
+function mapInfinityToNumber(value: number): number {
+	return value === Number.NEGATIVE_INFINITY ? -100 : value
+}
+
+/** Convert PAN value to string representation (L 100 .. C .. R 100) */
+function panValueToString(value: number): string {
+	if (value === 0) {
+		return 'L 100'
+	} else if (value === 1) {
+		return 'R 100'
+	} else if (value === 0.5) {
+		return 'C'
+	} else if (value > 0.5) {
+		const right = Math.round((value - 0.5) * 200)
+		return `R ${right}`
+	} else {
+		const left = Math.round((0.5 - value) * 200)
+		return `L ${left}`
+	}
+}
+
+export async function createVariables(
+	store: UiVariablesStore,
+	conn: SoundcraftUI,
+): Promise<CompanionVariableDefinition[]> {
+	const variables: CompanionVariableDefinition[] = []
+
+	// Helper to add variable and connect stream. Closure to capture `variables` and `store`
+	function addVar(data: { id: string; name: string }, stream: Observable<string | number>) {
+		variables.push({ variableId: data.id, name: data.name })
+		store.connect(data.id, stream)
+	}
+
+	const capabilities = await firstValueFrom(conn.deviceInfo.capabilities$)
```

Comment:
The mixer publishs its model after connection, and I use it to detect the device capabilities (number of channels, etc.).
Is it okay to create variable definitions asynchronously?